Islandora-Devops/claw-playbook
# Make playbook into two pieces

**GitHub Issue**: https://github.com/Islandora/documentation/issues/2166

# What does this Pull Request do?

This is a bit of a naive first attempt, probably needs a way to continue the original full build steps. Or if you wanted to use this to build a remote machine you'd need to run both sides.

For discussion.

# What's new?

Adds a new environment variable to check for called `ISLANDORA_BUILD_BASE` if it is "true" then the playbook will build a ubuntu/focal64 box with Java, Tomcat 9, Apache, MySQL/Postgres, ActiveMQ, Solr and Cantaloupe installed.

There is some configuration I didn't move, for instance Apache is configured to use the eventual Drupal folder as its DocumentRoot. But as there is nothing there you get a error page. 

* Does this change require documentation to be updated? Yes it will, if adopted
* Does this change add any new dependencies? no
* Does this change require any other modifications to be made to the repository
 (i.e. Regeneration activity, etc.)? no
* Could this change impact execution of existing code? 🤷 

# How should this be tested?

1. Pull down this PR in a directory.
2. Build the base box.
     ```
     > ISLANDORA_BUILD_BASE=true vagrant up
    ```
    This should import the `ubuntu/focal64` box, that will be one of the first steps it executes. Once it is complete (without errors 🤞 )
3. Package the box.
     ```
     > vagrant package --output islandora_base
     ```
     This shuts down the VM and exports it to the file `islandora_base` in this directory.
4. Get rid of the old box
     ```
     > vagrant destroy
          default: Are you sure you want to destroy the 'default' VM? [y/N] y
     ==> default: Forcing shutdown of VM...
     ==> default: Destroying VM and associated drives...
     ```
5. Now build a VM using the base box.
     ```
     > vagrant up
     ```
      This should show it is importing the `islandora_base` box.

That's it, you can now destroy and rebuild the VM using this base image as many times as you need.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora-Devops/committers especially @rosiel
